src/App.jsx

```diff
@@ -1,26 +1,30 @@
-import { useState } from 'react'
-import './css/App.css'
-import Header from './Header_and_Nav/Home/Header'
-import Nav from './Header_and_Nav/Home/Nav'
-import Home from './Home/Home'
-import { Route, Routes } from 'react-router-dom'
-import Topic from './topic/Topic'
-import Article from './Article/Article'
+import { useState } from "react";
+import "./css/App.css";
+import Header from "./Header_and_Nav/Home/Header";
+import Nav from "./Header_and_Nav/Home/Nav";
+import Home from "./Home/Home";
+import { Route, Routes } from "react-router-dom";
+import Topic from "./topic/Topic";
+import Article from "./Article/Article";
+import { ArticleContext } from "../Utils/articleContext";
 
 function App() {
-  const [topic, setTopic] = useState({})
+  const [topic, setTopic] = useState({});
+  const [articleid, setArticleID] = useState({});
 
   return (
     <>
-      <Header />
-      <Nav />
-      <Routes>
-        <Route path='/' element={<Home/>} />
-        <Route path={`/${topic}`} element={<Topic />} />
-        <Route path='/:article_id' element={<Article/>} />
-      </Routes>
+      <ArticleContext.Provider value={{ articleid, setArticleID }}>
```

Comment:
Why are we storing the Article ID in a context the article ID we change depending on the article we want to view.
And we could using something like useParams on a single article page to get the number we're looking for.